internal/http/services/owncloud/ocs/conversions/permissions.go

```diff
@@ -40,19 +40,21 @@ const (
 	PermissionShare
 	// PermissionAll grants all permissions on a resource
 	PermissionAll Permissions = (1 << (iota - 1)) - 1
+	// PermissionNone grants no permissions on a resource
+	PermissionNone = PermissionAll + 1
```

Comment:
`PermissionNone = 1 << PermisionAll`
provides a safer namespace

Reply:
Thanks for the review @labkode.
I think using my implementation is sufficient to ensure that all the bits of the previous permissions are zero, the only 1 bit is the one used only in PermissionNode.
It should work also for 1 << PermissionAll, but now PermissionAll = 31, so is a shift of 31 bits!! If you only add 2 more permissions you will overflow...

Reply:
I think the idea is to leave some "space" for potential future regular permissions, which would still be included in `PermissionAll` "before" `PermissioNone`. But yes, shifting it by 31 bits is way too much. Yet, we'd need to define `PermissionNone` _independently_ from `PermissionAll`, otherwise it would get redefined if another permission is added.

Is it an option to break now (but then be safe for the future) all permissions by having `PermissionNone = 1` and the others following? As well as defining `PermissionAll = (1 << (iota - 1)) - 2` i.e. all bits set except the None?

If that's not possible, we're left with e.g.:
`PermissionNone = 1 << 10     # hardcoded, 5 bits away from the currently used bits`
and the checks for valid values cannot use a `PermissionRead, PermissionNone` range as many values are undefined in between.

Reply:
Better yet:
```
PermissionShare
PermissionNone
PermissionMax Permissions = (1 << (iota - 1)) - 1
PermissionAll = PermissionMax - PermissionNone
```
Current values are all preserved, and validity checks can be based on the range `PermissionRead, PermissionMax`. `PermissionMax` is not to be used outside range checks.